rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m

```diff
@@ -0,0 +1,171 @@
+    [_reloadWebPageButton release];
+    [_closeWebPageButton release];
+    [_openInBrowserButton release];
+    [super dealloc];
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self setupLayout];
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+    self.navigationController.toolbarHidden = NO;
+}
+
+- (void)viewWillDisappear:(BOOL)animated {
+    [super viewWillDisappear:animated];
+    self.navigationController.toolbarHidden = YES;
+}
+
+// MARK: -
+
+- (void)setupLayout {
+    self.view.backgroundColor = UIColor.whiteColor;
+    [self.view addSubview:self.webView];
+    [NSLayoutConstraint activateConstraints:@[
+        [self.webView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
+        [self.webView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],
+        [self.webView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
+        [self.webView.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor]
+    ]];
+    
+    self.toolbarItems = @[
+        self.goBackButton,
+        [UIBarButtonItem fillerItem],
+        self.goForwardButton,
+        [UIBarButtonItem fillerItem],
+        self.reloadWebPageButton,
+        [UIBarButtonItem fillerItem],
+        self.closeWebPageButton,
+        [UIBarButtonItem fillerItem],
+        self.openInBrowserButton
+    ];
+}
+
+- (UIBarButtonItem *)webBarButtonItemWithImage:(UIImage *)image action:(SEL)selector {
+    return [[[UIBarButtonItem alloc] initWithImage:image
+                                             style:UIBarButtonItemStylePlain
+                                            target:self.webView
+                                            action:selector] autorelease];
+}
+
+// MARK: - Lazy
+
+- (WKWebView *)webView {
+    if(!_webView) {
+        _webView = [WKWebView new];
+        _webView.navigationDelegate = self;
+        _webView.translatesAutoresizingMaskIntoConstraints = NO;
+    }
+    return _webView;
+}
+
+- (UIBarButtonItem *)goBackButton {
+    if(!_goBackButton) {
+        _goBackButton = [[self webBarButtonItemWithImage:[UIImage imageNamed:@"arrow-narrow-left"]
+                                                  action:@selector(goBack)] retain];
+    }
+    return _goBackButton;
+}
+
+- (UIBarButtonItem *)goForwardButton {
+    if(!_goForwardButton) {
+        _goForwardButton = [[self webBarButtonItemWithImage:[UIImage imageNamed:@"arrow-narrow-right"]
+                                                     action:@selector(goForward)] retain];
+    }
+    return _goForwardButton;
+}
+
+- (UIBarButtonItem *)reloadWebPageButton {
+    if(!_reloadWebPageButton) {
+        _reloadWebPageButton = [[self webBarButtonItemWithImage:[UIImage imageNamed:@"refresh"]
+                                                         action:@selector(reload)] retain];
+    }
+    return _reloadWebPageButton;
+}
+
+- (UIBarButtonItem *)closeWebPageButton {
```

Comment:
Should be stop button (cancel load current url).
**Crash on tap.**

Reply:
fixed